js/fetcher.js

```diff
@@ -0,0 +1,36 @@
+import { getFromLocalStorage } from "./common/utils/localStorageUtil.js";
+
+// Checks if token is present
+const accessToken = getFromLocalStorage("accessToken");
+console.log("Access Token:", accessToken);
+
+export async function fetcher(
+  url,
+  options = { method: "POST" },
+  shouldUseAuth = false
+) {
+  try {
+    let fetchOptions = {
+      ...options,
+      headers: { "Content-Type": "application/json" },
+    };
+
+    //if Auth, add token to header
+    if (shouldUseAuth) {
+      const accessToken = getFromLocalStorage("accessToken");
+      fetchOptions = {
+        ...fetchOptions,
+        headers: {
+          ...fetchOptions.headers,
+          Authorization: `Bearer ${accessToken}`,
+        },
+      };
+    }
+
+    const response = await fetch(url, fetchOptions);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Error with API call", error);
```

Comment:
If you catch the error and just log it, the try/catch where you call the function (like in loginUser function, for example), will never receive the error in the catch block. I would remove the try/catch in this function.